opensearch-operator/api/v1/opensearch_types.go

```diff
@@ -119,6 +117,15 @@ type TlsSecret struct {
 	Key        *string `json:"key,omitempty"`
 }
 
+type SecurityConfig struct {
+	// Secret that contains the differnt yml files of the opensearch-security config (config.yml, internal_users.yml, ...)
+	SecurityconfigSecret corev1.LocalObjectReference `json:"securityConfigSecret,omitempty"`
+	// TLS Secret that contains a client certificate (tls.key, tls.crt, ca.crt) with admin rights in the opensearch cluster. Must be set if transport certificates are provided by user and not generated
+	AdminSecret corev1.LocalObjectReference `json:"adminSecret,omitempty"`
+	// Secret that contains fields username and password to be used by the operator to access the opensearch cluster for node draining. Must be set if custom securityconfig is provided.
+	AdminCredentialsSecret corev1.LocalObjectReference `json:"adminCredentialsSecret,omitempty"`
```

Comment:
what do you mean by ` access the opensearch cluster for node draining', why for node draining

Reply:
When the scaler drains a node it uses the API to exclude the node from hosting shards.  This requests cluster credentials.